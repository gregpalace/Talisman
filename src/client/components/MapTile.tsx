import React from 'react';

import { SPRITE_SIZE } from '../constants/actionTypes';
import { IMapTile } from '../../typings/MapTypes';

import '../../stylesheets/tiles.scss';

const MapTile: React.FC<IMapTile> = ({ tile }) => {
  const getTileSprite = (type: number | null) => {
    switch (type) {
      case 0:
        return 'grass';
      case 1:
        return 'tree';
      case 5:
        return 'rock';
      default:
        return null;
    }
  };

  return (
    <div
      className={`tile ${getTileSprite(tile)}`}
      style={{
        height: SPRITE_SIZE,
        width: SPRITE_SIZE,
      }}
    >
      {tile}
    </div>
  );
};

export default MapTile;
