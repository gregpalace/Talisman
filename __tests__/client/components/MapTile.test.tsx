import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MapTile from '../../../src/client/components/MapTile';

describe('React unit tests', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<MapTile />);
  });

  describe('<MapRow />', () => {
    it('Should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
