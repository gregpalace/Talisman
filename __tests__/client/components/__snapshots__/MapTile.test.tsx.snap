// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <MapRow /> Should render correctly 1`] = `
<div
  className="tile null"
  style={
    Object {
      "height": 50,
      "width": 50,
    }
  }
/>
`;
